bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+            }
+            var handler = thing.getHandler();
+            if (handler == null) {
+                logger.debug("No handler for thing {} refresh cancelled", thing.getUID());
+                continue;
+            }
+            var channels = thing.getChannels();
+            for (var channel : channels) {
+                handler.handleCommand(channel.getUID(), REFRESH);
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no commands in this bridge
+        logger.debug("Bridge does not support any commands to any channels. channelUID={}, command={}", channelUID,
+                command);
+    }
+
+    @Override
+    public void dispose() {
+        var localScheduledFuture = scheduledFuture;
+        if (localScheduledFuture != null) {
+            localScheduledFuture.cancel(true);
+            scheduledFuture = null;
+        }
+        super.dispose();
+    }
+
+    @Override
+    public SortedSet<@NonNull DeviceProperty<?>> findPropertiesForDevice(String dsn) {
+        return requireNonNullElse(devicePropertiesCache.get(dsn), emptySortedSet());
+    }
+
+    @Nullable
+    private SortedSet<DeviceProperty<?>> loadPropertiesForDevice(String dsn) {
+        var api = salusApi;
+        if (api == null) {
+            logger.error("Cannot find properties for device {} because salusClient is null", dsn);
+            return null;
+        }
+        logger.debug("Finding properties for device {} using salusClient", dsn);
+        var response = api.findDeviceProperties(dsn);
+        if (response.failed()) {
+            logger.error("Cannot find properties for device {} using salusClient\n{}", dsn, response.error());
+            return null;
+        }
+        return response.body();
+    }
+
+    @Override
+    public boolean setValueForProperty(String dsn, String propertyName, Object value) {
+        var api = salusApi;
+        if (api == null) {
+            logger.error("Cannot set value for property {} on device {} because salusClient is null", propertyName,
+                    dsn);
+            return false;
+        }
+        logger.debug("Setting property {} on device {} to value {} using salusClient", propertyName, dsn, value);
+        var response = api.setValueForProperty(dsn, propertyName, value);
+        if (response.failed()) {
+            logger.error("Cannot set property {} on device {} to value {} using salusClient\n{}", propertyName, dsn,
+                    value, response.error());
+            devicePropertiesCache.invalidate(dsn);
+            return false;
+        }
+        var setValue = response.body();
+        if (setValue != null
+                && (!(setValue instanceof Boolean) && !(setValue instanceof String) && !(setValue instanceof Number))) {
+            logger.warn(
+                    "Cannot set value {} ({}) for property {} on device {} because it is not a Boolean, String, Long or Integer",
```

Comment:
```suggestion
            logger.debug("Cannot set value {} ({}) for property {} on device {} because it is an unexpected type",
```

Reply:
are you sure it should be debug?

Reply:
From OH documentation (about warn logging):

> to inform the user that something seems to be wrong in the overall setup, but the system can nonetheless function as normal,

This log should never happen, because we know the type of channel a priori, but ... idk what might happen with the Salus API or other things that I do not have power on.

Reply:
We’ll leave it as is. Both variants have pros/cons. It’s easy to adapt in the future